File: trunk/C#/Sample Project/SampleProject/SampleProject.Operations/InstallCustomActions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.SharePoint;
using SampleProject.Common;

namespace SampleProject.Operations
{
    public class InstallCustomActions:BaseOperation
    {
        public InstallCustomActions(SPWeb objSPWeb) : base(objSPWeb) { }

        public void RegisterCustomCoreInScriptLink()
        {
            SPUserCustomAction objSPUserCustomActionScript = null;
            objSPUserCustomActionScript = this.CurrentWeb.UserCustomActions.Add();


            if (!CommonFunctions.IsExistsCustomAction("CustomCore.Js", "ScriptLink", String.Empty, this.CurrentWeb))
            {
                objSPUserCustomActionScript.Location = "ScriptLink";
                objSPUserCustomActionScript.Name = "CustomCore.Js";
                objSPUserCustomActionScript.ScriptSrc = "/_layouts/SampleProject/JS/CustomCore.js";
                this.CurrentWeb.AllowUnsafeUpdates = true;
                objSPUserCustomActionScript.Update();
            }
        }
    }
}

File: VB/Sample Project/SampleProject/SampleProject.Operations/InstallCustomActions.vb
Imports Microsoft.SharePoint
Imports SampleProject.Common

Public Class InstallCustomActions
    Inherits BaseOperation
    Public Sub New(ByVal objSPWeb As SPWeb)
        MyBase.New(objSPWeb)
    End Sub

    Public Sub RegisterCustomCoreInScriptLink()
        Dim objSPUserCustomActionScript As SPUserCustomAction = Nothing
        objSPUserCustomActionScript = Me.CurrentWeb.UserCustomActions.Add()

        If Not CommonFunctions.IsExistsCustomAction("CustomCore.Js", "ScriptLink", [String].Empty, Me.CurrentWeb) Then
            objSPUserCustomActionScript.Location = "ScriptLink"
            objSPUserCustomActionScript.Name = "CustomCore.Js"
            objSPUserCustomActionScript.ScriptSrc = "/_layouts/SampleProject/JS/CustomCore.js"
            Me.CurrentWeb.AllowUnsafeUpdates = True
            objSPUserCustomActionScript.Update()
        End If
    End Sub
End Class
